### DISCUSSION TITLE
Using Matrix for collaborative local-first editing

### DISCUSSION THREAD

**Discussion by VarLad** [fonsp/Pluto.jl#1986] [Pluto development] [answered]
Is there anyway we can use https://github.com/YousefED/Matrix-CRDT or https://www.npmjs.com/package/@collabs/collabs or directly using Matrix to make a local-first rich collaborative editor in Pluto?
I'd like to know the opinions regarding the feasibility of Matrix in doing rich collaborative development in Pluto.

**Comment by fonsp** [comment]
Pluto is already collaborative(!) in every sense except that _unsaved changes are not shared between clients (and server)_. https://pluto-with-friends.glitch.me/ 

We are working on #1729 which solves that!

What would be left after the PR is things like 3 people editing the cell at the same time, which would give some deleted code for 2 people. CRDT stuff would help, but we would just use the CM6 native solution: https://codemirror.net/6/examples/collab/

**Comment by fonsp** [comment, marked as answer]
You can use https://ngrok.io/ to share your local Pluto server with friends! But keep in mind that making your Pluto server available on the internet can be very insecure!
